packages/eas-cli/src/project/expoConfig.ts

```diff
@@ -40,12 +59,38 @@ function getExpoConfigInternal(
 }
 
 export function getPrivateExpoConfig(projectDir: string, opts: ExpoConfigOptions = {}): ExpoConfig {
+  ensureExpoConfigExists(projectDir);
+
   return getExpoConfigInternal(projectDir, { ...opts, isPublicConfig: false });
 }
 
+export function ensureExpoConfigExists(projectDir: string): void {
+  const paths = getConfigFilePaths(projectDir);
+  if (!paths?.staticConfigPath && !paths?.dynamicConfigPath) {
+    // eslint-disable-next-line node/no-sync
+    fs.writeFileSync(path.join(projectDir, 'app.json'), JSON.stringify({ expo: {} }, null, 2));
```

Comment:
I think `slug` and `name` are required to be considered valid (based on xdl schema)?

Reply:
fwiw `getConfig`from '@expo/config' will populate `slug` and `name` by default. For example, if I had an empty `app.json`, `getConfig()` will return smth like this:

```
{
  name: 'AwesomeProject2',
  slug: 'awesomeproject2',
...
}
```

As long as there is an `expo` field in `app.json`, any calls to `modifyConfig` should generate a correct `app.json` (there are some legacy reasons why this is the case)

Reply:
Interesting. I'm assuming that those fields are never written to the file, and instead just returned as part of the config from `getConfig`. This could have odd consequences if the user changes their directory name (it defaults to `path.basename(projectRoot)`); their project ID would now mismatch. Also we need to be sure that we only ever use `getConfig` to access the config and never upload it directly as there are definitely tools that require these fields to exist.

Reply:
yeah - i noticed t

> fwiw `getConfig`from '@expo/config' will populate `slug` and `name` by default. For example, if I had an empty `app.json`, `getConfig()` will return smth like this:
> 
> ```
> {
>   name: 'AwesomeProject2',
>   slug: 'awesomeproject2',
> ...
> }
> ```
> 
> As long as there is an `expo` field in `app.json`, any calls to `modifyConfig` should generate a correct `app.json` (there are some legacy reasons why this is the case)

yeah, i originally had implemented this by checking if `getConfig` returned a truthy value then i remembered this 😅 so i changed to getting config file paths instead.

> This could have odd consequences if the user changes their directory name (it defaults to path.basename(projectRoot)); their project ID would now mismatch

definitely! it's been like this for a while though so not going to try to solve it in this pr